workers/unity/Packages/io.improbable.gdk.testutils/MockCommandSender.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.TestUtils;
+using Unity.Entities;
+
+namespace Packages.io.improbable.gdk.testutils
+{
+    public class MockCommandSender : ICoreCommandSender
+    {
+        private readonly MockWorld world;
+        public long NextRequestId { get; private set; } = 1;
+
+        private readonly Dictionary<Type, uint> componentIds = new Dictionary<Type, uint>();
+
+        private readonly Dictionary<Type, SortedSet<long>> requestIds = new Dictionary<Type, SortedSet<long>>();
+
+        private readonly Dictionary<long, ICommandRequest> outboundRequests = new Dictionary<long, ICommandRequest>();
+
+        internal MockCommandSender(MockWorld world)
+        {
+            this.world = world;
+        }
+
+        public void Setup<TRequest>(uint componentId)
+        {
+            componentIds.Add(typeof(TRequest), componentId);
+        }
+
+        public IEnumerable<long> GetOutboundCommandRequestIds<TRequest>() where TRequest : ICommandRequest
+        {
+            return requestIds[typeof(TRequest)];
+        }
+
+        public CommandRequestId SendCommand<TRequest>(TRequest request, Entity sendingEntity = default) where TRequest : ICommandRequest
+        {
+            if (!requestIds.ContainsKey(typeof(TRequest)))
+            {
+                requestIds.Add(typeof(TRequest), new SortedSet<long>());
+            }
+
+            if (!componentIds.ContainsKey(typeof(TRequest)))
+            {
+                throw new AggregateException($"Could not retrieve Component ID of the {typeof(TRequest)} command");
+            }
+
+            requestIds[typeof(TRequest)].Add(NextRequestId);
+            outboundRequests.Add(NextRequestId, request);
+            return new CommandRequestId(NextRequestId++);
+        }
+
+        public void GenerateResponses<TRequest, TResponse>(Func<CommandRequestId, TRequest, TResponse> creator)
+            where TRequest : ICommandRequest
+            where TResponse : struct, IReceivedCommandResponse
+        {
+            var ids = requestIds[typeof(TRequest)];
+            var componentId = componentIds[typeof(TRequest)];
+            foreach (var id in ids)
+            {
+                var request = outboundRequests[id];
+                world.Connection.AddCommandResponse(creator(new CommandRequestId(id), (TRequest) request), componentId, (uint) id);
```

Comment:
I think you are passing in the wrong `id` here for the last parameter to `AddCommandResponse`. It should be `commandId` (which I believe is the schema command index of the command) rather than `requestId`.

Reply:
(Similarly below)